platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ExportMigrationConfigurationAdminContext.java

```diff
@@ -0,0 +1,208 @@
+     *
+     * @param cfg the config to check its validity
+     * @return <code>true</code> if the config is valid; <code>false</code> otherwise
+     */
+    private boolean isValid(Configuration cfg) {
+        final String fpid = Objects.toString(cfg.getProperties()
+                .get(ConfigurationAdmin.SERVICE_FACTORYPID), null);
+
+        if (fpid == null) {
+            return true;
+        } // else - property reports it shouldbe a managed service factory, so it is valid only if the cfg object reports it is too
+        return ConfigurationAdminMigratable.isManagedServiceFactory(cfg);
+    }
+
+    public Stream<ExportMigrationEntry> entries() {
+        return entries.values()
+                .stream();
+    }
+
+    @Override
+    public ExportMigrationEntry getEntry(Path path) {
+        throw new UnsupportedOperationException("should not be called");
+    }
+
+    @Override
+    public Stream<ExportMigrationEntry> entries(Path path) {
+        throw new UnsupportedOperationException("should not be called");
+    }
+
+    @Override
+    public Stream<ExportMigrationEntry> entries(Path path, PathMatcher filter) {
+        throw new UnsupportedOperationException("should not be called");
+    }
+
+    private ExportMigrationEntry getEntry(Configuration configuration) {
+        Path path = getPathFromConfiguration(configuration);
+        final String extn = FilenameUtils.getExtension(path.toString());
+        PersistenceStrategy ps = admin.getPersister(extn);
+
+        if (ps == null) {
+            ps = admin.getDefaultPersister();
+            if (warnedExtensions.add(extn)) {
+                getReport().record(new MigrationWarning(String.format(
+                        "Persistence strategy [%s] is not defined; defaulting to [%s]",
+                        extn,
+                        ps.getExtension())));
+            }
+            path = Paths.get(
+                    path.toString() + FilenameUtils.EXTENSION_SEPARATOR + ps.getExtension());
+        }
+        return new ExportMigrationConfigurationAdminEntry(super.getEntry(path), configuration, ps);
+    }
+
+    private Path getPathFromConfiguration(Configuration configuration) {
+        final Object o = configuration.getProperties()
+                .get(DirectoryWatcher.FILENAME);
+        Path path = null;
+
+        if (o != null) {
+            try {
+                if (o instanceof URL) {
+                    path = new File(((URL) o).toURI()).toPath();
+                } else if (o instanceof URI) {
+                    path = new File((URI) o).toPath();
+                } else if (o instanceof String) {
+                    path = new File(new URL((String) o).toURI()).toPath();
+                } else if (o instanceof File) {
+                    path = ((File) o).toPath();
+                } else if (o instanceof Path) {
+                    path = (Path) o;
+                } else {
+                    path = constructPathForBasename(configuration);
```

Comment:
❓ This code and the one in the `catch` block seem to be identical. Can we extract that to a `private` method?

Reply:
not quite identical, the logs and the recorded warnings are different. The common code was already abstracted out into the constructPathBasename() method